cuppa-core/src/main/java/org/forgerock/cuppa/internal/TestRunner.java

```diff
@@ -50,13 +56,18 @@ private void runTests(TestBlock testBlock, boolean ignoreTestsNotMarkedAsOnly, B
                 }
             }
             for (Test t : testBlock.tests) {
-                if (ignoreTestsNotMarkedAsOnly && combinedBehaviour.combine(t.behaviour) == only
-                        || !ignoreTestsNotMarkedAsOnly) {
-                    testWrapper.apply(() -> runTest(t, combinedBehaviour, reporter));
+                if (shouldRunTestBasedOnTag(testTags, t.tags, runTags)) {
+                    if (ignoreTestsNotMarkedAsOnly && combinedBehaviour.combine(t.behaviour) == only
+                            || !ignoreTestsNotMarkedAsOnly) {
+                        testWrapper.apply(() -> runTest(t, combinedBehaviour, reporter));
+                    }
                 }
             }
+            //@Checkstyle:off
```

Comment:
Checkstyle doesn't deal with indentation in lambdas correctly. Hopefully something that they will fix in a later version.